app/helpers/general-date.js

```diff
@@ -25,12 +26,14 @@ export function generalDate(params, { tz }) {
     format = format.replace(' a', '');
   }
 
-  let dateTime = moment(params[0]).tz(timezone).format(format);
-  const z = moment(params[0]).tz(timezone).format('z');
-  const reg = new RegExp('[+-]?[0-9]+');
-  
-  if (reg.test(z)) {
-    dateTime = dateTime.replace(z, timezone);
+  const formatWithoutTz = format.replace(' (z)', '');
+
+  let dateTime = moment(params[0]).tz(timezone).format(formatWithoutTz);
+
+  const timezoneAbbr = tzAbbr[timezone] || moment(params[0]).tz(timezone).format('z');
+
+  if (format.includes('(z)')) {
+    dateTime = dateTime.concat(` (${timezoneAbbr})`);
```

Comment:
Don't do this much of replacing and concat

Just remove (z) from the format literal, and append the abbreviation if the passed in code format was among the tz variants

Reply:
and then what about default case?

Reply:
I thought we moved all timezone displaying strings in keywords. That was the goal of this PR

Reply:
> I thought we moved all timezone displaying strings in keywords. That was the goal for this PR

oh really, if that's the case then I can do that

Reply:
Please check once as well

Reply:
goal for this or for the other timezone add argument tz one PR ?

Reply:
Done with all tz display string keywords